apps/src/dance/Dance.js

```diff
@@ -99,9 +99,12 @@ Dance.prototype.init = function (config) {
   const showFinishButton = !this.level.isProjectLevel && !this.level.validationCode;
   const finishButtonFirstLine = _.isEmpty(this.level.softButtons);
 
+  const songManifest = await getSongManifest(config.useRestrictedSongs);
```

Comment:
Do we really want to delay first render until the song manifest loads, or can this come in after?

Reply:
We can definitely accomplish rendering without waiting to fetch the song manifest and signed cookies, but it would require some refactoring because the song selector currently requires the manifest and selected song on initial render. The following things would probably need to be moved from SongSelector into redux to accomplish this: `parseSongOptions()`, `loadSong()`, `props.songManifest`, `state.songsData`. I don't mind doing this, but I'd recommend doing it in a separate PR.